src/main/java/net/dv8tion/jda/api/utils/FileUpload.java

```diff
@@ -358,6 +363,45 @@ public FileUpload setDescription(@Nullable String description)
         return this;
     }
 
+    /**
+     * Turns this attachment into a voice message with the provided waveform.
+     *
+     * @param  mediaType
+     *         The audio type for the attached audio file. Should be {@code audio/ogg} or similar.
+     * @param  waveform
+     *         The waveform of the audio, which is a low frequency sampling up to 256 bytes.
+     * @param  durationSeconds
+     *         The actual duration of the audio data in seconds.
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided or the waveform is not between 1 and 256 bytes long.
+     *
+     * @return The same FileUpload instance configured as a voice message attachment
+     */
+    @Nonnull
+    public FileUpload asVoiceMessage(@Nonnull MediaType mediaType, @Nonnull byte[] waveform, double durationSeconds)
+    {
+        Checks.notNull(mediaType, "Media type");
+        Checks.notNull(waveform, "Waveform");
+        Checks.check(waveform.length > 0 && waveform.length <= 256, "Waveform must be between 1 and 256 bytes long");
+        Checks.check(Double.isFinite(durationSeconds), "Duration must be a finite number");
+        Checks.check(durationSeconds > 0, "Duration must be positive");
+        this.waveform = waveform;
+        this.durationSeconds = durationSeconds;
+        this.mediaType = mediaType;
+        return this;
+    }
+
+    /**
+     * Whether this attachment is a valid voice message attachment.
+     *
+     * @return True, if this is a voice message attachment.
+     */
+    public boolean isVoiceMessage()
+    {
+        return this.mediaType.type().equals("audio");
```

Comment:
This should also check waveform and duration, to make sure this isn't true by mistake.